client/src/pages/Prospects.js

```diff
@@ -60,20 +62,75 @@ const useStyles = makeStyles((theme) => ({
 
 const Prospects = () => {
   const classes = useStyles();
+  const [prospects, handleProspects] = useState([]);
+  const [dialog, handleDialog] = useState(null);
+  const [campaigns, handleCampaigns] = useState(null);
+  const [campaignId, setCampaignId] = useState(null);
+
+  useEffect(() => {
+    getAllCampaigns();
+  }, [])
+
+  // Need to replace hard code user with JWT token 
+  const user = 1
+
+  const getAllCampaigns = () => {
+    fetch(`http://localhost:5000/campaigns/${user}`)
+    .then(res => res.json())
+      .then(data => {
+        handleCampaigns(data.Campaigns)
+    })
+    .catch(err => {
+      console.log(err.message);
+    });
+  }
+
+  const callSave = () => {
+    handleDialog(false)
+    saveProspectsToCampaign()
+  } 
+
+  const saveProspectsToCampaign = () => {
+    const data = {
+      "prospect_ids": prospects,
+    };
 
-  const prepareData = () => {
+    fetch(`http://localhost:5000/campaign/${campaignId}/prospects`, {
+      method: "POST",
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify(data)
+    })
+      
+    .then(res => res.json())
+      .then(data => {
+        return data.message;
+    })
+    .catch(err => {
+      console.log(err.message);
+    });
+  }
+
+  const addToCampaignButton = prospects.length > 0 ? <CustomizedButton onClick={() =>  handleDialog(true)} > Add to Campaign </CustomizedButton> : null;
+  const addToCampaignDialog = dialog === true ? <CustomizedDialog handleDialog={handleDialog} dialog={dialog} campaigns={campaigns} setCampaignId={setCampaignId} callSave={callSave}/> : null;
```

Comment:
might be easier to show both of these right in the render and to use something like this:
``{prospects.length > 0 && <CustomizedButton onClick={() =>  handleDialog(true)} > Add to Campaign </CustomizedButton>}``